kafka/producer/base.py

```diff
@@ -200,7 +253,16 @@ def _send_messages(self, topic, partition, *msg, **kwargs):
 
         if self.async:
             for m in msg:
-                self.queue.put((TopicAndPartition(topic, partition), m, key))
+                try:
+                    item = (TopicAndPartition(topic, partition), m, key)
+                    if self.async_queue_put_timeout == 0:
+                        self.queue.put_nowait(item)
+                    else:
+                        self.queue.put(item, True, self.async_queue_put_timeout)
+                except Full:
+                    raise AsyncProducerQueueFull(
```

Comment:
we'll need to have a way to let the user know which messages were successfully added to the queue, and which were not (because we loop `for m in msg` it is possible that the first n messages are added, but msg n+1 raises the exception.